### PR TITLE
OSGi features cleanup

### PR DESCRIPTION
**Pull request by neykov**
Depends on https://github.com/apache/brooklyn-server/pull/303

### CODE REVIEW

File: karaf/features/src/main/feature/feature.xml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        <feature>war</feature>
+    </feature>
+
+    <feature name="brooklyn-jsgui" version="${project.version}" description="Brooklyn REST JavaScript Web GUI">
+        <feature>brooklyn-jsgui-prereqs</feature>
+        <bundle>mvn:org.apache.brooklyn/brooklyn-jsgui/${project.version}/war</bundle>
+    </feature>
+
+    <feature name="brooklyn-boot" version="${project.version}" description="Bundles to start on boot">
+        <feature prerequisite="true">brooklyn-with-ui</feature>
```
Comment (duncangrant) [inline karaf/features/src/main/feature/feature.xml]: When I tried to build this I get:
```
[ERROR] Failed to execute goal org.apache.karaf.tooling:karaf-maven-plugin:4.0.4:assembly (default-assembly) on project apache-brooklyn: Unable to build assembly: Could not find matching feature for brooklyn-with-ui -> [Help 1]
```

Comment (neykov) [inline karaf/features/src/main/feature/feature.xml]: Sorry about that, didn't push the latest changes. Should be all good now.

**Comment by geomacy** [comment]
Looks good to me from review point of view and AMP starts successfully. Haven't tested a large number of restarts using the script though.

**Comment by Graeme-Miller** [comment]
successfully tested

**Pull request merged**